### PR TITLE
feat: placeholders for unimplemented filters and support for Web

### PR DESCRIPTION
**Pull request by jakex7**
# Summary

This PR adds placeholders for all filters to enable their use on the Web. Users can import and use them on the Web; however, on native platforms, these filters will not have any effect and will display a warning.

### CODE REVIEW

File: src/elements/filters/FeColorMatrix.tsx [hunk 1]
```diff
@@ -6,13 +6,13 @@ import {
...
 } from '../../lib/extract/extractFilter';
-import { FilterColorMatrixType } from '../../lib/extract/types';
+import { FilterColorMatrixType, NumberArray } from '../../lib/extract/types';
 import FilterPrimitive from './FilterPrimitive';
 
 export type FeColorMatrixProps = {
   in?: string;
   type?: FilterColorMatrixType;
-  values?: number | Array<number> | string;
+  values?: NumberArray;
```
Comment (WoLewicki) [inline src/elements/filters/FeColorMatrix.tsx:15]: Why did it end up in this PR?

File: src/elements/filters/FeGaussianBlur.tsx [hunk 2]
```diff
@@ -6,12 +6,12 @@ import {
   extractFilter,
   extractIn,
 } from '../../lib/extract/extractFilter';
-import { FilterEdgeMode, NumberProp } from '../../lib/extract/types';
+import { FilterEdgeMode, NumberArray } from '../../lib/extract/types';
```
Comment (WoLewicki) [inline src/elements/filters/FeGaussianBlur.tsx:9]: Same here

File: src/lib/extract/extractFilter.ts [hunk 3]
```diff
@@ -46,7 +46,9 @@ export const extractFeColorMatrix = (
 
   if (props.values !== undefined) {
     if (Array.isArray(props.values)) {
-      extracted.values = props.values;
+      extracted.values = props.values.map((num) =>
```
Comment (WoLewicki) [inline src/lib/extract/extractFilter.ts:49]: Why do we do those changes in this PR?

File: src/lib/extract/types.ts [hunk 4]
```diff
@@ -9,6 +9,7 @@ import type React from 'react';
 
 export type NumberProp = string | number;
 export type NumberArray = NumberProp[] | NumberProp;
+export type BooleanProp = boolean | 'true' | 'false';
```
Comment (WoLewicki) [inline src/lib/extract/types.ts:12]: Why can it be string too?

File: src/xmlTags.ts [hunk 5]
```diff
@@ -30,18 +44,34 @@ import {
   TSpan,
   Use,
 } from './elements';
+import FeDistantLight from './elements/filters/FeDistantLight';
```
Comment (WoLewicki) [inline src/xmlTags.ts:47]: Why only this is imported in a different style?

**Review by WoLewicki — COMMENTED** [review]
Left some comments.

File: src/xmlTags.ts (hunk 5 above)
Comment (jakex7) [reply src/xmlTags.ts:48]: Fixed in [b3fbd3c](https://github.com/software-mansion/react-native-svg/pull/2392/commits/b3fbd3c56649435baf1b59531623a17275ddd12d)

File: src/lib/extract/types.ts (hunk 4 above)
Comment (jakex7) [reply src/lib/extract/types.ts:12]: Exactly same reason as NumberProp can be string. 
Example component should be valid:
```tsx
  <Filter id="convolveMatrix2" x="0" y="0" width="100%" height="100%">
    <FeConvolveMatrix
      kernelMatrix="1 -1 2 0 0 0 0 0 -2"
      preserveAlpha="false" />
  </Filter>
```

File: src/elements/filters/FeGaussianBlur.tsx (hunk 2 above)
Comment (jakex7) [reply src/elements/filters/FeGaussianBlur.tsx]: Reverted in [31af51b](https://github.com/software-mansion/react-native-svg/pull/2392/commits/31af51bf062a2054b5082d2693829a2f7b3e46a4)

File: src/elements/filters/FeColorMatrix.tsx (hunk 1 above)
Comment (jakex7) [reply src/elements/filters/FeColorMatrix.tsx]: Reverted in [31af51b](https://github.com/software-mansion/react-native-svg/pull/2392/commits/31af51bf062a2054b5082d2693829a2f7b3e46a4)

File: src/lib/extract/extractFilter.ts (hunk 3 above)
Comment (jakex7) [reply src/lib/extract/extractFilter.ts]: Reverted in [31af51b](https://github.com/software-mansion/react-native-svg/pull/2392/commits/31af51bf062a2054b5082d2693829a2f7b3e46a4)

**Pull request merged**